doc/sslang-lrm.tex

```diff
@@ -328,6 +325,77 @@ \subsection{User-defined operators}
 a look at Agda's user-defined operators.
 
 \section{Types}
+sslang exposes two classes of types. \\
+
+\noindent The first class of types encompasses primitive types: integers, booleans, pure events (i.e. conditional variable), and unit (i.e. void).\\ 
+
+\noindent The second class of types are higher order types that are composed of primitive types:
+\begin{itemize}
+    \item \texttt{[<type>]}: An array containing values of type \texttt{<type>}
+    \item \texttt{(<type>, ...)}: A tuple containing values of type \texttt{<type>}
+    \item \texttt{<type1> $\rightarrow$ <type2>}: A function that takes a   \texttt{<type1>} and returns a  \texttt{<type2>}
+    \item \texttt{\&<type>}: A reference to a variable of type \texttt{<type>}
+    \item \texttt{Sched <type>}: A scheduled variable of type \texttt{<type>}. These variables are scheduled and managed by the SSM runtime and are used to synchronize routines
+\end{itemize}
+
+\noindent This is summarized by the following grammar:
+\setlength{\grammarindent}{5em}
+\begin{grammar}
+<type> ::= [<type>]
+\alt (<type>(, <type>)+)
+\alt <type> $\rightarrow$ <type>
+\alt \&<type>
+\alt Sched <type>
+\alt (<type>)
+\alt Int8 | Int16 | Int32 | Int64
```

Comment:
These type names should all be encompassed under a single rule for typenames (and the same should go for `Sched`).

So something like
```
<type> ::= [<type>]
    | (<type>(, <type>)+)
    | <type> -> <type>
    | &<type>
    | (<type>)
    | ()
    | <type-id>

<type-id> ::= [A-Z][a-zA-Z0-9]*
```

Reply:
Yup! I realized I'd have to rewrite this section after the Slack convo today.